airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/state_converters/abstract_stream_state_converter.py

```diff
@@ -19,11 +19,60 @@ class AbstractStreamStateConverter(ABC):
     END_KEY = "end"
 
     @abstractmethod
+    def _from_state_message(self, value: Any) -> Any:
+        pass
+
+    @abstractmethod
+    def _to_state_message(self, value: Any) -> Any:
+        pass
+
+    def __init__(self, is_sequential_state: bool = True):
+        self._is_sequential_state = is_sequential_state
+
+    def convert_to_state_message(self, cursor_field: "CursorField", stream_state: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
```

Comment:
We now see that `cursor_field` is only relevant when it comes to sequential state and hence it should probably not be part of the interface

Reply:
too bad that we need to break the interface, but I'm def. in favor of aggressively removing parameters that are not needed

Reply:
We would need to extract the sequential part from the converter in order to do that. I'm not willing to take that much time in tackling debt that is not blocking this change